[0.71][Android] Update pages on New Architecture for 71 (#3453)
facebook/react-native-website: docs/the-new-architecture/pillars-fabric-components.md

A Fabric Native Component is created starting from a **JavaScript specification**. Then [**Codegen**](./pillars-codegen) creates some C++ scaffolding code to connect the component-specific logic (for example, accessing some native platform capability) to the rest of the React Native infrastructure. The C++ code is the same for all the platforms. Once the component is properly connected with the scaffolding code, it is ready to be imported and used by an app.

> The following section guides you through the creation of a Fabric Native Component, step-by-step, targeting React Native 0.71.0.

Reviewer: ```suggestion
The following section guides you through the creation of a Fabric Native Component, step-by-step, targeting the latest version of React Native.
```

All the requirements for the New Architecture have been encapsulated in the [`install_modules_dependencies`](https://github.com/facebook/react-native/blob/82e9c6ad611f1fb816de056ff031716f8cb24b4e/scripts/react_native_pods.rb#L145). It takes care of installing the proper dependencies based on which architecture is currently enabled. It also automatically installs the `React-Core` dependency in the old architecture.

> ### Android: `build.gradle` and the `ReactPackage` class

Reviewer: Isn't the `AndroidManaifest.xml` required anymore?

> To prepare Android to run **Codegen** you have to create two files:

Reviewer: ```suggestion
To prepare Android to run **Codegen** you have to:
```

> 1. The `build.gradle` with the **Codegen** configuration

Reviewer: ```suggestion
1. Update the `build.gradle` file.
```

> 2. A Java/Kotlin class that implements the `ReactPackage` interface.

Reviewer: ```suggestion
2. Create a Java/Kotlin class that implements the `ReactPackage` interface.
```
